Bottom: 09e1fe8c92f5ecb3210d11fee8f5f9099ae272df
Top:    108ab6afd619538c4707d9fcc73acb14eb29fbcf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:45:50 +0000

Refresh of file-c-cleanup-mixed-pclose-fc

---

diff --git a/src/file.c b/src/file.c
index d649515..f71d1b0 100644
--- a/src/file.c
+++ b/src/file.c
@@ -1539,10 +1539,11 @@ ReadNetlist (char *filename)
   LibraryEntryTypePtr entry;
   int i, j, lines, kind;
   bool continued;
-  int used_popen = 0;
+  bool used_popen = false;
+  int retval = 0;
 
   if (!filename)
-    return (1);			/* nothing to do */
+    return 1;			/* nothing to do */
 
   Message (_("Importing PCB netlist %s\n"), filename);
 
@@ -1557,7 +1558,7 @@ ReadNetlist (char *filename)
     }
   else
     {
-      used_popen = 1;
+      used_popen = true;
       free (command);
       command = EvaluateFilename (Settings.RatCommand,
 				  Settings.RatPath, filename, NULL);
@@ -1566,7 +1567,7 @@ ReadNetlist (char *filename)
       if (*command == '\0' || (fp = popen (command, "r")) == NULL)
 	{
 	  PopenErrorMessage (command);
-	  return (1);
+	  return 1;
 	}
     }
   lines = 0;
@@ -1637,15 +1638,14 @@ ReadNetlist (char *filename)
   if (!lines)
     {
       Message (_("Empty netlist file!\n"));
-      pclose (fp);
-      return (1);
+      retval = 1;
     }
   if (used_popen)
     pclose (fp);
   else
     fclose (fp);
   sort_netlist ();
-  return (0);
+  return retval;
 }
 
 static int ReadEdifNetlist (char *filename);
